tests/unit-tests/gconstruct/test_transform.py

```diff
@@ -85,6 +85,16 @@ def test_fp_transform(input_dtype):
     assert_equal(max_val[0], 5.)
     assert_equal(min_val[0], -5.)
 
+    feats = np.random.randn(100).astype(input_dtype)
+    feats[0] = 10.
+    feats[1] = -10.
+    transform = NumericalMinMaxTransform("test", "test")
+    max_val, min_val = transform.pre_process(feats)["test"]
+    max_v = np.amax(feats).astype(np.float32)
+    min_v = np.amin(feats).astype(np.float32)
+    assert len(max_val.shape) == 1
+    assert len(min_val.shape) == 1
+
```

Comment:
You need to test the case when the out_dtype is float16, but you give the max_bound = np.finfo(float32).max, then check whether the self._max_bound == np.finfo(float16).max or not.